kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/bluetooth/le/BluetoothLeAdapter.java

```diff
@@ -0,0 +1,241 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Eurotech and/or its affiliates
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Eurotech
+ *******************************************************************************/
+package org.eclipse.kura.bluetooth.le;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.Future;
+import java.util.function.Consumer;
+
+import org.eclipse.kura.KuraException;
+import org.osgi.annotation.versioning.ProviderType;
+
+/**
+ * BluetoothAdapter represents the physical Bluetooth adapter on the host machine (ex: hci0).
+ *
+ * @noimplement This interface is not intended to be implemented by clients.
+ */
+@ProviderType
+public interface BluetoothLeAdapter {
+
+    /**
+     * Search for a BLE device with the specified address. The method will perform a BLE discovery for at most timeout
+     * seconds or until the device is found. It will return a Future instance and the discovered device can be
+     * retrieved using the get() method. The get(long timeout, TimeUnit unit) is not supported and acts as the get()
+     * method.
+     * 
+     * @param timeout
+     *            timeout in seconds for device discovery
+     * @param address
+     *            MAC address of the BLE device
+     * @return Future
+     */
+    public Future<BluetoothLeDevice> findDeviceByAddress(long timeout, String address);
```

Comment:
We can think about this. IMHO, it is better to offer more flexibility to the user. But this can be more complex and less user-friendly.